bindings/java/CMakeLists.txt

```diff
@@ -256,29 +254,81 @@ if(NOT OPEN_FOR_IDE)
-    add_jar(fdb-junit SOURCES ${JAVA_JUNIT_TESTS} INCLUDE_JARS fdb-java ${CMAKE_BINARY_DIR}/packages/junit-4.13.jar)
+  endif()
+
+  if(RUN_JUNIT_TESTS)
+    # Sets up the JUnit testing structure to run through ctest
+    #
+    # To add a new junit test, add the class to the JAVA_JUNIT_TESTS variable in `src/tests.cmake`. Note that if you run a Suite, 
+    # ctest will NOT display underlying details of the suite itself, so it's best to avoid junit suites in general. Also,
+    # if you need a different runner other than JUnitCore, you'll have to modify this so be aware.
+    #
+    # To run tests (once built), run:
+    #
+    # ctest .
+    #
+    # from the ${BUILD_DIR}/bindings/java subdirectory
+    #
+    # ctest has lots of flexible command options, so be sure to refer to its documentation if you want to do something specific(documentation
+    # can be found at https://cmake.org/cmake/help/v3.19/manual/ctest.1.html)
+
+    add_jar(fdb-junit SOURCES ${JAVA_JUNIT_TESTS} ${JUNIT_RESOURCES} INCLUDE_JARS fdb-java ${CMAKE_BINARY_DIR}/packages/junit-4.13.jar ${CMAKE_BINARY_DIR}/packages/hamcrest-all-1.3.jar)
     get_property(junit_jar_path TARGET fdb-junit PROPERTY JAR_FILE)
-    add_test(NAME junit
-      COMMAND ${Java_JAVA_EXECUTABLE}
-              -cp "${target_jar}:${junit_jar_path}:${CMAKE_BINARY_DIR}/packages/junit-4.13.jar:${CMAKE_BINARY_DIR}/packages/hamcrest-all-1.3.jar"
-              -Djava.library.path=${CMAKE_BINARY_DIR}/lib
-              org.junit.runner.JUnitCore "com.apple.foundationdb.tuple.AllTests")
+    
+    foreach(JTEST IN LISTS JAVA_JUNIT_TESTS)
+            string(REPLACE "/" "." TEST_CLASS ${JTEST})
+            string(REGEX REPLACE ".java$" "" TEST_CLASS ${TEST_CLASS})
+            string(REGEX REPLACE "^src.junit." "" TEST_CLASS ${TEST_CLASS})
+            add_test(NAME ${TEST_CLASS}
+        COMMAND ${Java_JAVA_EXECUTABLE}
+                -cp "${target_jar}:${junit_jar_path}:${CMAKE_BINARY_DIR}/packages/junit-4.13.jar:${CMAKE_BINARY_DIR}/packages/hamcrest-all-1.3.jar"
+                -Djava.library.path=${CMAKE_BINARY_DIR}/lib
+                org.junit.runner.JUnitCore "${TEST_CLASS}"
+        )
+    endforeach()
+  endif()
+
+  if(RUN_JAVA_INTEGRATION_TESTS)
+    # Set up the integration tests. These tests generally require a running database server to function properly
+    #
+    # Integration tests are run through ctest the same way as unit tests, but their label is prefixed with the entry 'integration-'.
+    # Note that most java integration tests will be skipped if they can't quickly connect to a running FDB instance(depending on how the test is written, anyway). 
+    # However, if you want to explicitly skip them, you can run 
+    #
+    # `ctest -E integration`
+    #
+    # To run only integration tests, execute
+    #
+    # `ctest -R integration`
+    #
+    # (Note: both of these commands must be run from ${BUILD_DIR}/bindings/java)
```

Comment:
Why is that? Shouldn't it be fine to run this in the build directory? I

Reply:
It is fine, it's just not as pleasant an experience for the pure java dev. Will update the documentation with alternative approaches as well.